// preact-material-components/TabBar/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TabBar = exports.Tab = exports.TabIcon = exports.TabLabel = void 0;

var _get2 = _interopRequireDefault(require("@babel/runtime/helpers/get"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _tabBar = require("@material/tab-bar");

var _preact = require("preact");

var _MaterialComponent5 = _interopRequireDefault(require("../Base/MaterialComponent"));

var TabLabel =
/*#__PURE__*/
function (_MaterialComponent) {
  (0, _inherits2.default)(TabLabel, _MaterialComponent);

  function TabLabel() {
    var _this;

    (0, _classCallCheck2.default)(this, TabLabel);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(TabLabel).apply(this, arguments));
    _this.componentName = 'tab__text-label';
    _this.mdcProps = [];
    return _this;
  }

  (0, _createClass2.default)(TabLabel, [{
    key: "materialDom",
    value: function materialDom(props) {
      return (0, _preact.h)("span", Object.assign({}, props), props.children);
    }
  }]);
  return TabLabel;
}(_MaterialComponent5.default);

exports.TabLabel = TabLabel;

var TabIcon =
/*#__PURE__*/
function (_MaterialComponent2) {
  (0, _inherits2.default)(TabIcon, _MaterialComponent2);

  function TabIcon() {
    var _this2;

    (0, _classCallCheck2.default)(this, TabIcon);
    _this2 = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(TabIcon).apply(this, arguments));
    _this2.componentName = 'tab__icon';
    _this2.mdcProps = [];
    return _this2;
  }

  (0, _createClass2.default)(TabIcon, [{
    key: "materialDom",
    value: function materialDom(props) {
      return (0, _preact.h)("span", Object.assign({
        className: "material-icons"
      }, props), props.children);
    }
  }]);
  return TabIcon;
}(_MaterialComponent5.default);

exports.TabIcon = TabIcon;

var Tab =
/*#__PURE__*/
function (_MaterialComponent3) {
  (0, _inherits2.default)(Tab, _MaterialComponent3);

  function Tab() {
    var _this3;

    (0, _classCallCheck2.default)(this, Tab);
    _this3 = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Tab).apply(this, arguments));
    _this3.componentName = 'tab';
    _this3.mdcProps = ['active'];
    _this3.mdcNotifyProps = ['active'];
    return _this3;
  }

  (0, _createClass2.default)(Tab, [{
    key: "materialDom",
    value: function materialDom(props) {
      return (0, _preact.h)("button", Object.assign({
        class: "mdc-tab",
        role: "tab",
        "aria-selected": "true"
      }, props), (0, _preact.h)("span", {
        class: "mdc-tab__content"
      }, props.children), (0, _preact.h)("span", {
        class: "mdc-tab-indicator ".concat(props.active ? 'mdc-tab-indicator--active' : '')
      }, (0, _preact.h)("span", {
        class: "mdc-tab-indicator__content mdc-tab-indicator__content--underline"
      })), (0, _preact.h)("span", {
        class: "mdc-tab__ripple"
      }));
    }
  }]);
  return Tab;
}(_MaterialComponent5.default);

exports.Tab = Tab;

var TabBar =
/*#__PURE__*/
function (_MaterialComponent4) {
  (0, _inherits2.default)(TabBar, _MaterialComponent4);

  function TabBar() {
    var _this4;

    (0, _classCallCheck2.default)(this, TabBar);
    _this4 = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(TabBar).apply(this, arguments));
    _this4.componentName = 'tab-bar';
    _this4.mdcProps = [];
    _this4.mdcNotifyProps = ['activeTabIndex'];
    return _this4;
  }

  (0, _createClass2.default)(TabBar, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      (0, _get2.default)((0, _getPrototypeOf2.default)(TabBar.prototype), "componentDidMount", this).call(this);

      if (this.control) {
        this.MDComponent = new _tabBar.MDCTabBar(this.control);
      }

      this.afterComponentDidMount();
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      (0, _get2.default)((0, _getPrototypeOf2.default)(TabBar.prototype), "componentWillUnmount", this).call(this);

      if (this.MDComponent) {
        this.MDComponent.destroy();
      }
    }
  }, {
    key: "materialDom",
    value: function materialDom(props) {
      return (0, _preact.h)("div", {
        class: "mdc-tab-bar",
        role: "tablist",
        ref: this.setControlRef
      }, (0, _preact.h)("div", {
        class: "mdc-tab-scroller"
      }, (0, _preact.h)("div", {
        class: "mdc-tab-scroller__scroll-area"
      }, (0, _preact.h)("div", {
        class: "mdc-tab-scroller__scroll-content"
      }, props.children))));
    }
  }]);
  return TabBar;
}(_MaterialComponent5.default);

exports.TabBar = TabBar;

var default_1 =
/*#__PURE__*/
function (_TabBar) {
  (0, _inherits2.default)(default_1, _TabBar);

  function default_1() {
    (0, _classCallCheck2.default)(this, default_1);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(default_1).apply(this, arguments));
  }

  return default_1;
}(TabBar);

exports.default = default_1;
default_1.Tab = Tab;
default_1.TabLabel = TabLabel;
default_1.TabIcon = TabIcon;
