// preact-material-components/packages/top-app-bar/lib/index.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { MDCTopAppBar } from '@material/top-app-bar';
import { MaterialComponent } from '@preact-material-components/base/lib/MaterialComponent';
import { bind } from 'bind-decorator';
import { h } from 'preact';
export class TopAppBar extends MaterialComponent {
    constructor() {
        super(...arguments);
        this.componentName = 'top-app-bar';
        this.mdcProps = ['short', 'short-collapsed', 'fixed', 'prominent'];
    }
    componentDidMount() {
        super.componentDidMount();
        if (this.control) {
            const comp = new MDCTopAppBar(this.control);
            comp.listen('MDCTopAppBar:nav', this.onNav);
            this.MDComponent = comp;
        }
    }
    componentWillUnmount() {
        super.componentWillUnmount();
        if (this.MDComponent) {
            this.MDComponent.unlisten('MDCTopAppBar:nav', this.onNav);
            this.MDComponent.destroy();
        }
    }
    onNav(e) {
        if (this.props.onNav) {
            this.props.onNav(e);
        }
    }
    materialDom(props) {
        return (h("header", Object.assign({ ref: this.setControlRef }, props), props.children));
    }
}
__decorate([
    bind
], TopAppBar.prototype, "onNav", null);
