// preact-material-components/packages/tab/lib/index.js
import { MDCTabBar } from '@material/tab-bar';
import { MaterialComponent } from '@preact-material-components/base/lib/MaterialComponent';
import { h } from 'preact';
export class TabBar extends MaterialComponent {
    constructor() {
        super(...arguments);
        this.componentName = 'tab-bar';
        this.mdcProps = [];
        this.mdcNotifyProps = ['activeTabIndex'];
    }
    componentDidMount() {
        super.componentDidMount();
        if (this.control) {
            this.MDComponent = new MDCTabBar(this.control);
        }
        this.afterComponentDidMount();
    }
    componentWillUnmount() {
        super.componentWillUnmount();
        if (this.MDComponent) {
            this.MDComponent.destroy();
        }
    }
    materialDom(props) {
        return (h("div", { class: "mdc-tab-bar", role: "tablist", ref: this.setControlRef },
            h("div", { class: "mdc-tab-scroller" },
                h("div", { class: "mdc-tab-scroller__scroll-area" },
                    h("div", { class: "mdc-tab-scroller__scroll-content" }, props.children)))));
    }
}
