// preact-material-components/LineRipple/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.LineRipple = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _get2 = _interopRequireDefault(require("@babel/runtime/helpers/get"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _lineRipple = require("@material/line-ripple");

var _preact = require("preact");

var _MaterialComponent2 = _interopRequireDefault(require("../Base/MaterialComponent"));

var LineRipple =
/*#__PURE__*/
function (_MaterialComponent) {
  (0, _inherits2.default)(LineRipple, _MaterialComponent);

  function LineRipple() {
    var _this;

    (0, _classCallCheck2.default)(this, LineRipple);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(LineRipple).apply(this, arguments));
    _this.componentName = 'line-ripple';
    _this.mdcProps = [];
    return _this;
  }

  (0, _createClass2.default)(LineRipple, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      (0, _get2.default)((0, _getPrototypeOf2.default)(LineRipple.prototype), "componentDidMount", this).call(this);

      if (this.control) {
        this.MDComponent = new _lineRipple.MDCLineRipple(this.control);
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      (0, _get2.default)((0, _getPrototypeOf2.default)(LineRipple.prototype), "componentWillUnmount", this).call(this);

      if (this.MDComponent) {
        this.MDComponent.destroy();
      }
    }
  }, {
    key: "materialDom",
    value: function materialDom(props) {
      return (0, _preact.h)("div", Object.assign({}, props, {
        ref: this.setControlRef
      }));
    }
  }]);
  return LineRipple;
}(_MaterialComponent2.default);

exports.LineRipple = LineRipple;
var _default = LineRipple;
exports.default = _default;
